bootstrap.php

```diff
@@ -78,51 +78,106 @@ public function dismissibleMessage($text)
 END;
   }
 
+  public function getAuthUrl() {
+    return "https://wordpress_auth.intercom.io/confirm?state=".get_site_url()."::".wp_create_nonce('intercom-oauth');
+  }
+
   public function htmlUnclosed()
   {
     $settings = $this->getSettings();
     $app_id = WordPressEscaper::escAttr($settings['app_id']);
     $secret = WordPressEscaper::escAttr($settings['secret']);
+    $secure_mode = WordPressEscaper::escAttr($settings['secure_mode']);
+    $auth_url = $this->getAuthUrl();
 
-    if (empty($secret)) {
-      $secret_row_style = 'display: none;';
-      $secret_link_style = '';
+    if($secure_mode) {
+      $secure_mode_state = "checked disabled";
+    }
+    if (empty($app_id)) {
+      $app_id_row_style = 'display: none;';
+      $app_id_link_style = '';
     } else {
-      $secret_row_style = '';
-      $secret_link_style = 'display: none;';
+      $app_id_row_style = '';
+      $app_id_link_style = 'display: none;';
     }
 
     $dismissable_message = '';
     if ($_GET['saved']) {
-      $dismissable_message = $this->dismissibleMessage('App ID saved.');
+      $dismissable_message = $this->dismissibleMessage('Successfully authenticated');
     }
-
-    if ($_GET['appId']) {
-      $app_id = WordPressEscaper::escAttr($_GET['appId']);
-      $dismissable_message = $this->dismissibleMessage('We’ve copied your new Intercom app ID below. Click to save changes and then close this window to finish signing up for Intercom.');
+    if ($_GET['enable_secure_mode']) {
+      $dismissable_message = $this->dismissibleMessage('Secure Mode successfully enabled');
     }
 
     return <<<END
-<div class="wrap">
```

Comment:
Can we double check that no unescaped content is going into the HTML? PHP won't automatically escape like rails :)